pandas/plotting/_matplotlib/core.py

```diff
@@ -411,11 +413,22 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError("no numeric data to plot")
 
+        def convert_to_ndarray(data):
```

Comment:
we do almost exactly this here: https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L2054

what we need is a conversion routine in pandas/core/dtypes/cast.py which consolidates these types of things

you can call it `safe_convert_to_ndarray`

Reply:
Added `safe_convert_to_ndarray`.